src/modal/modal.js

```diff
@@ -521,6 +521,16 @@ angular.module('ui.bootstrap.modal', ['ui.bootstrap.stackedMap', 'ui.bootstrap.p
         }
 
         appendToElement.addClass(modalBodyClass);
+        if (modal.scope) {
+          // ngStyle is associated with the modal scope and because the "index"
+          // property (which is private to the isolated scope of the uibModalWindow directive)
+          // is undefined here, it fails to compute the zIndex property.
+          // So we add it explicitly to the scope.
+          // This might need to be marked as a breaking change because
+          // "$$topModalIndex" could eventually shadow a property with the
+          // same name defined on the provided modal scope.
+          modal.scope.$$topModalIndex = topModalIndex;
```

Comment:
i'm ok w/ the name seeing as one is _not_ supposed to be using `$` as prefixes for their code.  but this is good insight to be aware of.